File: cpp/1068.cpp
#include <iostream>
#include <string>

using namespace std;

bool balance(string e)
{
    int a = 0, b = 0, i;

    for (i = 0; i < e.length(); i++) {
        if (e[i] == '(') {
            a++;
        } else if (e[i] == ')') {
            b++;

            if (a < b) {
                return false;
            }
        }
    }

    return (a == b) ? true : false;
}

int main()
{
    string expression;

    while (getline(cin, expression)) {
        if (balance(expression)) {
            cout << "correct\n";
        } else {
            cout << "incorrect\n";
        }
    }

    return 0;
}

File: py/1068.py
def balance(e):
    a, b = 0, 0

    for i in e:
        if i == '(':
            a += 1

        elif i == ')':
            b += 1

            if a < b:
                return False

    if a == b:
        return True

    return False


while True:
    try:
        expression = input()

        if balance(expression):
            print('correct')
        else:
            print('incorrect')

    except EOFError:
        break